### PR TITLE
added newrelic close buffer

### PR DESCRIPTION
**Pull request by armstrmi**
## Description of Changes
* Added close buffer functionality newrelic
* Added tests to test buffer functionality
## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

File: operator/builtin/output/newrelic/newrelic.go [hunk 1]
```diff
@@ -180,64 +186,12 @@ func (nro *NewRelicOutput) feedFlusher(ctx context.Context) {
 		case errors.Is(err, context.Canceled):
 			return
 		case err != nil:
-			nro.Errorf("Failed to read chunk", zap.Error(err))
+			nro.flusher.Errorf("Failed to read chunk", zap.Error(err))
```
Comment (BinaryFissionGames) [inline operator/builtin/output/newrelic/newrelic.go:189]: I think we should revert this line, doesn't make sense to use the flusher's logger here.

File: operator/builtin/output/newrelic/newrelic.go [hunk 2]
```diff
@@ -147,7 +139,21 @@ func (nro *NewRelicOutput) Stop() error {
...
 	nro.flusher.Stop()
 	// TODO deal with buffer Drain
-	_, err := nro.buffer.Close()
+	entries, err := nro.buffer.Close()
+	if err != nil {
+		return fmt.Errorf("failed to close buffer: %w", err)
+	}
+
+	if len(entries) != 0 {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
```
Comment (BinaryFissionGames) [inline operator/builtin/output/newrelic/newrelic.go:148]: The timeout here should be 5 seconds to keep consistency with other operators

File: operator/builtin/output/newrelic/newrelic.go [hunk 3]
```diff
@@ -156,21 +162,21 @@ func (nro *NewRelicOutput) Process(ctx context.Context, entry *entry.Entry) erro
...
-	res, err := nro.client.Do(req)
-	if err != nil {
-		return err
-	}
+	return nil
+}
 
-	return nro.handleResponse(res)
+func (nro *NewRelicOutput) testConnection() error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
```
Comment (BinaryFissionGames) [inline operator/builtin/output/newrelic/newrelic.go:176]: timeout here should be `nro.timeout` to keep old behaviour

File: operator/builtin/output/newrelic/client.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	if err := c.checkResponse(res); err != nil {
+		return err
+	}
+
+	return nil
```
Comment (BinaryFissionGames) [inline operator/builtin/output/newrelic/client.go:55]: linter error here that needs to be fixed

**Review by BinaryFissionGames — COMMENTED** [review]
Got a few nitpicks, looks good!

**Comment by BinaryFissionGames** [comment]
The windows unit test failure is interesting, https://github.com/golang/go/issues/22553 - may or may not be related to this pr, I'd be a little surprised if it was related, though.

**Review by cpheps — APPROVED** [review]

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**